fileblock/src/inode.rs

```diff
@@ -0,0 +1,17 @@
+
+/// an description of the type of file object pointed to by an Inode
+enum FileType {
+    /// a regular file
+    RegularFile,
+    /// a directory tree containing 1 or more regular files or directories
+    Directory
+}
+
+/// index node containing most of the interesting bits about a file object on disk
+struct Inode {
+    inumber: u32,
+    ftype: FileType,
+    fsize: u64,
+    block_location_id: u32,
```

Comment:
From what I've read it's standard to include a fixed array of block ids for storage here rather than using a linked list type structure (e.g. [u32;<SOME FIXED AMOUNT OF BLOCKS>]). This does two things:

1. allows for concurrent block reads/writes
2. simplifies reading for sizes of BLOCK_SIZE * number of block slots

Reply:
ah no I think you're on to something different -- the `incode.c` file linked there is what's described in ostep as a multi-index design IIUC. 

I like both of those points and I thnk it's simpler having a nice array